### ISSUE TITLE
Implement light app bar

### ISSUE THREAD

**Issue by knjk04** [Project-Books/book-project#625]
**Describe the solution you'd like**
- Add a material UI switch that lets a user toggle dark mode (by default, we want light mode)
![image](https://user-images.githubusercontent.com/11173328/107140710-e01d9280-691b-11eb-826a-d416cbfdafb4.png)

- Add a light app bar with a shadow. The logo, material UI icons and text should be white in light mode
![image](https://user-images.githubusercontent.com/11173328/107140761-3d194880-691c-11eb-9174-d8d14e19f96b.png)

**Comment by RivaD2** [comment]
@knjk04 I will take this one if that's cool. I am kinda already familiar with the app bar from the last issue I worked on :)

**Comment by knjk04** [comment]
@RivaD2 Perfect, thanks!

**Comment by AkshatGadhwal** [comment]
Hi @knjk04,  can I contribute to this issue?

**Comment by knjk04** [comment]
@AkshatGadhwal Sorry, I've already assigned Riva to it. Feel free to request to be assigned on an unblocked and unassigned issue

**Comment by RivaD2** [comment]
@knjk04 Here is my approach: 
- **Step 1:** Create the Material-UI Switch component using the Basic Switch option(default color, grey) as seen here: 
[Material-UI Switch](https://material-ui.com/components/switches/#switch)

- **Step 2**: Import the Switch component into the `Settings.tsx` file for use

- **Step 3:** Create a method called `toggleTheme` in `Settings.tsx` component and pass `onClick` as a prop to the `Switch` component to trigger the DarkMode toggle when `Switch` is clicked
- I will most likely achieve toggling dark mode/ setting light mode as default by using `CreateMuiTheme` and `ThemeProvider` OR by using Material-UI palette types which might be handy, as white is the default palette as seen here:
[Material-UI Palette](https://material-ui.com/customization/palette/)

That is my thought process so far,  I welcome any feedback you have:)

**Comment by knjk04** [comment]
@RivaD2 Thanks for describing your approach! It sounds good to me. I'd recommend creating a theme/using a theme provider so that we can better tailor it to our needs (if I've understood the documentation correctly). Thanks!

**Comment by RivaD2** [comment]
@knjk04 Where should this preference for light or dark mode be stored? So I started to work and now I see that it could be stored in the `App.tsx` state and triggered by a callback prop in `Settings.tsx` OR it of course could be stored elsewhere like in Redux, Context API etc...What are your thoughts here? 

Thank you for your help!

**Comment by knjk04** [comment]
@RivaD2 I'm quite new to React and I've never used Redux or the Context API before, so I'll have to look into those. 

For now, I'm thinking we should store it in `App.tsx` as a temporary approach. Eventually, it would be good if we could persist a user's preferences (there may be things besides light/dark mode) in our database (via our backend)

**Comment by RivaD2** [comment]
@knjk04 Thank you for your input, it is much appreciated!

**Comment by RivaD2** [comment]
@knjk04 I am pretty much done :) The remaining part is getting the image sizing/letter orientation for black logo to match that of the white logo. 
I compared the provided black logo to the light logo and they are different as far as indentation/text direction go. For the white logo, the letters are stacked in a column, for black they are in a row. 
Can you provide a bit of direction for me on how I might change the .png to get the letters stacked?

Thank you for your help!

**Comment by knjk04** [comment]
@RivaD2 Great!

![logo-two-lines-black@1x](https://user-images.githubusercontent.com/11173328/108810269-aba80880-75a2-11eb-90b8-6f1a44c5d32a.png)
![image](https://user-images.githubusercontent.com/11173328/108810298-be224200-75a2-11eb-8dac-c0c38b51aac7.png)

> I compared the provided black logo to the light logo and they are different as far as indentation/text direction go. For the white logo, the letters are stacked in a column, for black they are in a row.

Sorry, what do you mean by this? I thought they were both stacked in a column?

**Comment by RivaD2** [comment]
@knjk04 I see what is going on, my apologies. I was using the black logo from the `media/logo` folder rather than the black logo you shared with me in my open PR. The one in the `media/logo` folder is this:

![Screen Shot 2021-02-22 at 11 00 13 PM](https://user-images.githubusercontent.com/67027945/108811680-85fb0f00-7562-11eb-9137-3f5ca2d324e2.png)

I will make the change and we should be good to go.

**Comment by knjk04** [comment]
@RivaD2 No, I appreciate it wasn't clear. Perhaps we need to create subfolders in `media`. We use the one line black logo on the login screen and the one with two lines in the app bar.

The one with the PR, please. Thanks!

**Comment by RivaD2** [comment]
@knjk04 Alrighty!  I made the requested changes and added in an updated movie to the PR to show alll changes made. Hopefully I got it mostly right this time around:)

**Comment by RivaD2** [comment]
@knjk04 It seems that since the PR's are linked to issues, and we use the `closes #` syntax, that the issue should close automatically once PR is merged. In this case, since I can't close it, would you mind closin' this one? Thank you :)

**Comment by knjk04** [comment]
@RivaD2 I think they're not closing automatically because we didn't merge into the default branch (`main` in our case). I'll close this (thanks for the reminder!). Thanks again for all of the help!
